keywords/SyncGateway.py

```diff
@@ -285,3 +285,93 @@ def restart_sync_gateways(self, cluster_config, url=None):
+        playbook_vars = {}
+
+        sync_gateway_base_url, sync_gateway_package_name, sg_accel_package_name = sg_config.sync_gateway_base_url_and_package()
+
+        playbook_vars["couchbase_sync_gateway_package_base_url"] = sync_gateway_base_url
+        playbook_vars["couchbase_sync_gateway_package"] = sync_gateway_package_name
+        playbook_vars["couchbase_sg_accel_package"] = sg_accel_package_name
+
+        if url is not None:
+            target = hostname_for_url(cluster_config, url)
+            log_info("Upgrading sync_gateway/sg_accel on {} ...".format(target))
+            status = ansible_runner.run_ansible_playbook(
+                "upgrade-sync-gateway-package.yml",
+                subset=target,
+                extra_vars=playbook_vars
+            )
+            log_info("Completed upgrading {}".format(url))
+        else:
+            log_info("Upgrading all sync_gateways/sg_accels")
+            status = ansible_runner.run_ansible_playbook(
+                "upgrade-sync-gateway-package.yml",
+                extra_vars=playbook_vars
+            )
+            log_info("Completed upgrading all sync_gateways/sg_accels")
+        if status != 0:
+            raise Exception("Could not upgrade sync_gateway/sg_accel")
+
+    def enable_import_xattrs(self, cluster_config, sg_conf, url, enable_import=False):
+        """Deploy an SG config with xattrs enabled
+            Will also enable import if enable_import is set to True"""
+        ansible_runner = AnsibleRunner(cluster_config)
+        server_port = 8091
+        server_scheme = "http"
+
+        if is_cbs_ssl_enabled(cluster_config):
+            server_port = 18091
+            server_scheme = "https"
+
+        # Shared vars
+        playbook_vars = {
+            "sync_gateway_config_filepath": sg_conf,
+            "server_port": server_port,
+            "server_scheme": server_scheme,
+            "autoimport": "",
+            "xattrs": ""
+        }
+
+        if is_xattrs_enabled(cluster_config):
+            playbook_vars["xattrs"] = '"enable_shared_bucket_access": true,'
+
+        if is_xattrs_enabled(cluster_config) and enable_import:
+            playbook_vars["autoimport"] = '"import_docs": "continuous",'
+
+        # Deploy config
+        if url is not None:
+            target = hostname_for_url(cluster_config, url)
+            log_info("Deploying sync_gateway config on {} ...".format(target))
+            status = ansible_runner.run_ansible_playbook(
+                "deploy-sync-gateway-config.yml",
+                subset=target,
+                extra_vars=playbook_vars
+            )
+        else:
+            log_info("Deploying config on all sync_gateways")
+            status = ansible_runner.run_ansible_playbook(
+                "deploy-sync-gateway-config.yml",
+                extra_vars=playbook_vars
+            )
+        if status != 0:
+            raise Exception("Could not deploy config to sync_gateway")
```

Comment:
Parametrization is only for test cases. If you are referring to start_sync_gateways(), it does a lot more than just deploying a config. This method just deploys an SG config, it does not stop or start SG. This functionality is needed for the upgrade test. It would make the start_sync_gateways() very difficult to read.